monad-rpc/src/handlers/eth/txn.rs

```diff
@@ -230,45 +231,135 @@ pub async fn monad_eth_sendRawTransaction(
-                Ok(Ok(tx_status)) => match tx_status {
-                    TxStatus::Evicted { reason: _ } => {
-                        return Err(JsonRpcError::custom("rejected".to_string()))
-                    }
-                    TxStatus::Dropped { reason } => {
-                        return Err(JsonRpcError::custom(reason.as_user_string()))
-                    }
-                    TxStatus::Tracked | TxStatus::Committed => return Ok(hash.to_string()),
-                    TxStatus::Unknown => {
-                        error!("txpool bridge sent unknown status");
-                    }
-                },
-                Ok(Err(_)) | Err(_) => {
-                    warn!("txpool not responding");
+/// Polls for transaction receipt with timeout
+async fn poll_for_receipt<T: Triedb>(
+    chain_state: &ChainState<T>,
+    tx_hash: FixedBytes<32>,
+    timeout_ms: u64,
+    poll_interval_ms: u64,
+) -> Result<TransactionReceipt, JsonRpcError> {
+    let start_time = tokio::time::Instant::now();
+    let timeout = Duration::from_millis(timeout_ms);
+    let poll_interval = Duration::from_millis(poll_interval_ms);
+
+    loop {
+        match chain_state.get_transaction_receipt(*tx_hash).await {
+            Ok(receipt) => return Ok(receipt),
+            Err(ChainStateError::ResourceNotFound) => {
+                // Not found yet, check timeout
+                if start_time.elapsed() >= timeout {
+                    // EIP-7966: Error code 4 with tx hash in data
+                    return Err(JsonRpcError::tx_sync_timeout(tx_hash.to_string(), timeout_ms));
                 }
-            }
 
-            return Err(JsonRpcError::custom("txpool not responding".to_string()));
-        }
-        Err(e) => {
-            debug!(?e, "eth txn decode failed");
-            Err(JsonRpcError::txn_decode_error())
+                tokio::time::sleep(poll_interval).await;
+            }
+            Err(ChainStateError::Archive(e)) => {
+                return Err(JsonRpcError::internal_error(format!("Archive error: {e}")));
+            }
+            Err(ChainStateError::Triedb(e)) => {
+                return Err(JsonRpcError::internal_error(format!("Triedb error: {e}")));
+            }
         }
     }
 }
 
+#[rpc(
+    method = "eth_sendRawTransactionSync",
+    ignore = "txpool_bridge_client,chain_state,chain_id,allow_unprotected_txs"
+)]
+#[allow(non_snake_case)]
+#[tracing::instrument(level = "debug", skip_all)]
+pub async fn monad_eth_sendRawTransactionSync<T: Triedb>(
+    txpool_bridge_client: &EthTxPoolBridgeClient,
+    chain_state: &ChainState<T>,
+    params: MonadEthSendRawTransactionSyncParams,
+    chain_id: u64,
+    allow_unprotected_txs: bool,
+) -> JsonRpcResult<MonadTransactionReceipt> {
+    trace!("monad_eth_sendRawTransactionSync: {params:?}");
+
+    let timeout_ms = params.timeout_ms
+    .filter(|&t| t > 0 && t <= RECOMMENDED_TIMEOUT_MS)
+    .unwrap_or(RECOMMENDED_TIMEOUT_MS);
```

Comment:
We probably want to allow operators to set the maximum timeout allowed. See `monad-rpc/src/cli.rs`